lib/IRGen/GenEnum.cpp

```diff
@@ -3509,7 +3509,7 @@ namespace {
 
       if (AllowFixedLayoutOptimizations && TIK >= Loadable) {
         // The type layout entry code does not handle spare bits atm.
-        return IGM.typeLayoutCache.getOrCreateScalarEntry(getTypeInfo(), T);
+        return IGM.typeLayoutCache.getOrCreateResilientEntry(T);
```

Comment:
This adds the overhead of loading the value witness in code emission of e.g `copy_addr`. Something like:

```
  copy_addr %3 to [initialization] %2 : $*MultipayloadEnum  
```

Is going to call to the value witness functions (which requires loading the value witness table).


I think, instead we should introduce something like a `TypeInfoBasedTypeLayoutEntry` which is implemented like `ScalarTypeLayoutEntry` before this patch. We can use it until `ScalarTypeLayoutEntry` grows the full ability of TypeInfo based lowering (e.g handle spare bits computations).